wallet/src/de/schildbach/wallet/service/TxDisplayCacheService.kt

```
@@ -926,6 +966,64 @@ class TxDisplayCacheService @Inject constructor(
+
+    /**
+     * Cold-start variant: prefers the in-memory bitmap, then falls back to reading the
+     * `icon_bitmaps` table directly (the metadata flow may not have populated [metadata]
+     * yet when the fast-startup cache is first rendered).
+     */
+    private suspend fun iconBitmapForEntryCold(entry: TxDisplayCacheEntry): Bitmap? {
+        val iconId = entry.customIconId ?: return null
+        iconBitmapForEntry(entry)?.let { return it }
+        return try {
+            metadataProvider.getIcon(Sha256Hash.wrap(iconId))
+        } catch (e: IllegalArgumentException) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)